shelley/chain-and-ledger/dependencies/non-integer/doc/non-integer-calculations.tex

```diff
@@ -425,10 +471,16 @@ \section{Optimisations for Specific Use-Cases}
   \end{array}
 \end{equation*}
 
+\noindent
 The term $\exp(-\sigma \cdot c)$ can be computed using the Taylor series as
-described in Section~\ref{sec:taylor-series}.
-
-As the relevant information is not the result of the calculation, but whether
+described in Section~\ref{sec:taylor-series}. \khcomment{This can be simplified
+  further by applying $ln$ to each side to give $\ln(\frac{1}{q}) < (-\sigma \cdot c)$ and then
+  cancelling the negations to give $\ln(q) > \sigma \cdot c$.  This can then be used with
+  table lookup on $q$ to give a quick check as to whether the term might be true, and only applying
+  the complex calculation if so.  This should be more efficient than
+the calculation below, since most checks will fail.}
+
```

Comment:
we could also add the idea of using a linear lower (which we already have) and upper (which we should be able to define) bound for over- and under-approximation.

Reply:
Agreed!